src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -154,65 +166,31 @@ private DocumentBuilder getDocumentBuilder() {
     private void initializeAppsLocation(CommonLoggerI log, File serverXML, File configDir, File bootstrapFile,
             Map<String, String> bootstrapProp, File serverEnvFile, boolean giveConfigDirPrecedence, Map<String, File> libertyDirPropertyFiles) {
         try {
-            this.log = log;
-            serverXMLFile = serverXML;
-            configDirectory = configDir;
-            if (libertyDirPropertyFiles != null) {
-                libertyDirectoryPropertyToFile = new HashMap(libertyDirPropertyFiles);
-            } else {
-                log.warn("The properties for directories are null and could lead to application locations not being resolved correctly.");
-                libertyDirectoryPropertyToFile = new HashMap<String,File>();
-            }
-    
-            locations = new HashSet<String>();
-            names = new HashSet<String>();
-            namelessLocations = new HashSet<String>();
-            locationsAndNames = new HashMap<String, String>();
-            props = new Properties();
-            defaultProps = new Properties();
+            initializeFields(log, serverXML, configDir, libertyDirPropertyFiles);
 
             Document doc = parseDocument(serverXMLFile);
 
-            // Server variable precedence in ascending order if defined in
-            // multiple locations.
-            //
-            // 1. defaultValue from variables defined in server.xml or defined in <include/> files
-            // e.g. <variable name="myVarName" defaultValue="myVarValue" />
-            // 2. variables from 'server.env'
-            // 3. variables from 'bootstrap.properties'
-            // 4. variables defined in <include/> files
-            // 5. variables from configDropins/defaults/<file_name>
-            // 6. variables defined in server.xml
-            // e.g. <variable name="myVarName" value="myVarValue" />
-            // 7. variables from configDropins/overrides/<file_name>
+            // Server variable precedence in ascending order if defined in multiple locations.
+            //  1. variable default values in the server.xml file
+            //  2. environment variables
+            //  3. bootstrap.properties
+            //  4. Java system properties
+            //  5. Variables loaded from files in the ${server.config.dir}/variables directory or other directories as specified by the VARIABLE_SOURCE_DIRS environment variable
+            //  6. variable values declared in the server.xml file
```

Comment:
Why have we lost the detail of where configDropins fit?

Reply:
I think there is some confusion between two different lists on this [page](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html). There is a list for file processing [here](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#configuration-files). Then there is a list of variable substitution precedence [here](https://openliberty.io/docs/latest/reference/config/server-configuration-overview.html#variable-substitution). We need the comments here to kind of merge both of those into what the code logic should do.